components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/PasswordGrantHandler.java

```diff
@@ -112,6 +137,102 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx)
         return true;
     }
 
+    private AuthenticatedUser authenticateUserAtUserStore(OAuth2AccessTokenReqDTO tokenReq, String userId,
+                                                          AbstractUserStoreManager userStoreManager,
+                                                          String tenantAwareUserName)
+            throws org.wso2.carbon.user.core.UserStoreException {
+
+        AuthenticationResult authenticationResult;
+
+        if (userId != null) {
+            authenticationResult = userStoreManager.authenticateWithID(userId, tokenReq.getResourceOwnerPassword());
+        } else {
+            authenticationResult = userStoreManager.authenticateWithID(
+                    UserCoreClaimConstants.USERNAME_CLAIM_URI, tenantAwareUserName,
+                    tokenReq.getResourceOwnerPassword(), UserCoreConstants.DEFAULT_PROFILE);
+        }
+        boolean authenticated = AuthenticationResult.AuthenticationStatus.SUCCESS
+                == authenticationResult.getAuthenticationStatus()
+                && authenticationResult.getAuthenticatedUser().isPresent();
+        if (log.isDebugEnabled()) {
+            log.debug("user " + tokenReq.getResourceOwnerUsername() + " authenticated: " + authenticated);
+        }
+        if (authenticated) {
+            return new AuthenticatedUser(authenticationResult.getAuthenticatedUser().get());
```

Comment:
why are we creating a new authenticated user object here without passig the one coming from the authentication result?

Reply:
This is the existing code and I have moved to the separate method. The created **AuthenticatedUser** model is defined at framework level and the authenticatedUser model returned from **authenticationResult.getAuthenticatedUser().get()** is from userStore. Hence, the conversion of kernel level model to framework level model happens with this.